workflow/controller/workflowpod.go

```diff
@@ -753,7 +762,7 @@ func addVolumeReferences(pod *apiv1.Pod, vols []apiv1.Volume, tmpl *wfv1.Templat
 		return nil
 	}
 
-	if tmpl.Container != nil {
+	if tmpl.Container != nil && !isImpliedPod {
```

Comment:
==== `&& tmpl.Type != wfv1.TemplateTypeData ` ????

Reply:
This was part of my previous implementation. Since the PR is in a Draft state, I was planning on reviewing all the changes before expecting any review. I would have caught this then